pkg/process/checks/net.go

```diff
@@ -126,15 +130,96 @@ func (c *ConnectionsCheck) Realtime() bool { return false }
 // ShouldSaveLastRun indicates if the output from the last run should be saved for use in flares
 func (c *ConnectionsCheck) ShouldSaveLastRun() bool { return false }
 
+func (c *ConnectionsCheck) handleBatch(batch *model.Connections, start time.Time, groupID int32) (model.MessageBody, error) {
+	err := c.processData.Fetch()
+	if err != nil {
+		log.Warnf("error collecting processes for filter and extraction: %s", err)
+	} else {
+		c.dockerFilter.Filter(batch)
+	}
+	// Resolve the Raddr side of connections for local containers
+	LocalResolver.Resolve(batch)
+
+	c.notifyProcessConnRates(c.config, batch)
+
+	log.Debugf("collected connections in %s", time.Since(start))
+
+	res := processBatchGRPC(c.hostInfo, c.maxConnsPerMessage, groupID, batch.Conns, batch.Dns, c.networkID, batch.ConnTelemetryMap, batch.CompilationTelemetryByAsset, batch.KernelHeaderFetchResult, batch.CORETelemetryByAsset, batch.PrebuiltEBPFAssets, batch.Domains, batch.Routes, batch.Tags, batch.AgentConfiguration, c.serviceExtractor)
+	return res, nil
+
+}
```

Comment:
you never return error from that function, so you can remove the `error` from the returned list